app/bundles/UserBundle/Security/Authentication/Token/PluginToken.php

```diff
@@ -29,12 +30,21 @@ public function __construct(
             throw new \InvalidArgumentException('$providerKey must not be empty.');
         }
 
+        if (is_string($user)) {
+            $this->authenticatingUser = $user;
+            $user                     = null;
+        }
+
         if (null !== $user) {
             $this->setUser($user);
         }
-        $this->providerKey           = $providerKey;
 
-        $this->setAuthenticated(count($roles) > 0);
+        $this->providerKey = $providerKey;
+    }
+
+    public function getAuthenticatingUser(): ?string
```

Comment:
I can't find any place where this method is used?

Reply:
Was made to avoid BC breaks later in the code, but was not needed in the end. Removed.